internal/metamorph/processor.go

```diff
@@ -384,6 +394,37 @@ func (p *Processor) StartSendStatusUpdate() {
 	}()
 }
 
+func (p *Processor) StartCheckingTransactionsInNetwork() {
+	p.waitGroup.Add(1)
+	ticker := time.NewTicker(100 * time.Millisecond)
```

Comment:
Open question: Is it necessary to lock the `announcedTransactions` collection every 100 milliseconds? A better approach might be to increase the interval to 0.5-1 second and decrease the delta between the current time and the transaction announcement time. WDYT?

Reply:
Honestly not sure what we need here exactly. Or in other words what's the best time interval to wait before requesting transaction.